src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/NonDivArithmeticOpAnalyzer.php

```diff
@@ -401,6 +401,24 @@ private static function analyzeNonDivOperands(
             return null;
         }
 
+        if ($left_type_part instanceof Type\Atomic\TNumeric
+            && ($right_type_part instanceof TInt || $right_type_part instanceof TFloat)
+        ) {
```

Comment:
Hmm, doesn't it need to also handle operands in reverse order? It looks like it only works for `numeric OP int|float` but not for `int|float OP numeric`

Reply:
This shouldn't have mattered because we're in a double loop: https://github.com/vimeo/psalm/pull/6167/files/e65e0603543b8aa7b7c8f25f6bb6126ab81ad522#diff-465b13d382a5e0d258a3a7a124a668c181b5268daa281d08fe75730eddc44914L182

But there was another piece of code that was messing with this one. I realized my fix was wrong and it broke the checks for InvalidOperand in strictBinaryMode that is handled below. I refactored the code, let's see if tests pass.